src/socket.rs

```diff
@@ -369,6 +369,47 @@ impl Socket {
         self.inner.set_ttl(ttl)
     }
 
+    /// Gets the value of the `TCP_MAXSEG` option on this socket.
+    ///
+    /// The `TCP_MAXSEG` option denotes the TCP Maximum Segment
+    /// Size and is only available on TCP sockets.
+    pub fn mss(&self) -> io::Result<u32> {
```

Comment:
The CI fails on Windows, let's just mark it as unix only (`cfg[unix]`) for now and we'll figure out Windows 10 support when someone requests it.